### PR TITLE
LH-12: DC, Cloud; Edit issue description

### PR DESCRIPTION
**Pull request by ysemenkov**

### CODE REVIEW

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt [hunk 1]
```diff
@@ -28,6 +31,31 @@ class CloudIssuePage(
         }
     }
 
+    override fun editDescription(description: String): CloudIssuePage {
+        driver
+            .findElement(By.cssSelector("div[data-test-id = 'issue.views.field.rich-text.description']"))
```
Comment (dagguh) [inline custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]: I'd avoid requiring a `div`, we don't really mind if it's a `div` or `span` or whatever.
```suggestion
            .findElement(By.cssSelector("[data-test-id = 'issue.views.field.rich-text.description']"))
```

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt [hunk 2]
```diff
@@ -28,6 +31,31 @@ class CloudIssuePage(
...
+            .wait(
+                ExpectedConditions.presenceOfElementLocated(By.cssSelector("div[data-test-id='issue.views.field.rich-text.editor-container']"))
+            )
+
+        Actions(driver)
+            .sendKeys(description)
+            .perform()
+
+        descriptionForm
+            .findElement(By.cssSelector("button[data-testid='comment-save-button']"))
```
Comment (dagguh) [inline custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]: Does it work? I'd expect this:
```suggestion
            .findElement(By.cssSelector("[data-test-id='comment-save-button']"))
```

**Review by dagguh — CHANGES_REQUESTED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/action/WorkAnIssue.kt [hunk 3]
```diff
@@ -49,6 +49,9 @@ class WorkAnIssue(
     }
 
     private fun edit(issuePage: AbstractIssuePage) {
+        meter.measure(ActionType("Edit Issue Description") {Unit}) {
+            issuePage.editDescription("updated")
+        }
```
Comment (dagguh) [inline custom-vu/src/main/kotlin/jces1209/vu/action/WorkAnIssue.kt:50]: You can validate how it looks like thanks to `TakeScreenshotHook`. It should generate a sample of screenshots per every `ActionType`. AFAIK it's only used in `JiraCloudScenario` right now, so you can include it in `JiraDcScenario` too.

File: custom-vu/src/main/kotlin/jces1209/vu/action/WorkAnIssue.kt [hunk 4]
```diff
@@ -49,6 +49,9 @@ class WorkAnIssue(
     }
 
     private fun edit(issuePage: AbstractIssuePage) {
```
Comment (dagguh) [inline custom-vu/src/main/kotlin/jces1209/vu/action/WorkAnIssue.kt:47]: In order for this to be used, we gotta bump `editProbability` in `ScenarioSimilarities`

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt (hunk 2 above)
Comment (ysemenkov) [reply custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]: `Save` button has the attribute without the dash but i'll remove `button` tag from the selector  
  
`<button type="button" data-testid="comment-save-button" class="css-1yx6h60">...</button>`

```suggestion
            .findElement(By.cssSelector("[data-testid='comment-save-button']"))
```

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt [hunk 5]
```diff
@@ -28,6 +31,31 @@ class CloudIssuePage(
...
     }
 
+    override fun editDescription(description: String): CloudIssuePage {
+        driver
+            .findElement(By.cssSelector("[data-test-id = 'issue.views.field.rich-text.description']"))
+            .click();
+
+        val descriptionForm = driver
+            .wait(
+                ExpectedConditions.presenceOfElementLocated(By.cssSelector("div[data-test-id='issue.views.field.rich-text.editor-container']"))
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]:
```suggestion
                ExpectedConditions.presenceOfElementLocated(By.cssSelector("[data-test-id='issue.views.field.rich-text.editor-container']"))
```

**Review by dagguh — APPROVED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt (hunk 2 above)
Comment (dagguh) [reply custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]: Whooooah, nice find!

**Pull request merged**